vermouth/tests/test_ff_files.py

```diff
@@ -0,0 +1,1019 @@
+                """
+                [ impropers ]
+                BB SC1 SC2 SC3
+                """,
+                (),
+            ),
+            (
+                """
+                [ dihedrals ]
+                BB SC1 SC2 SC3 2 ; gromacs dihedrals of type 2 are impropers
+                """,
+                (),
+            ),
+            (  # the edge section creates edges
+                """
+                [ edges ]
+                SC1 SC2
+                BB SC3
+                """,
+                (('SC1', 'SC2'), ('BB', 'SC3')),
+            ),
+            (  # edges can be deactivated from interactions
+                """
+                [ bonds ]
+                BB SC1 -- {"edge": false}
+                SC2 SC3 -- {"edge": true}
+                """,
+                (('SC2', 'SC3'), ),
+            ),
+    ))
+    def test_interaction_edges(interaction_lines, edges):
+        """
+        Edges are created where expected.
+        """
+        lines = """
+        [ moleculetype ]
+        GLY 1
+
+        [ atoms ]
+        1 P4 1 ALA BB 1
+        2 P3 1 ALA SC1 2
+        3 P2 1 ALA SC2 3
+        4 P2 1 ALA SC3 3
+        """
+        lines = textwrap.dedent(lines) + textwrap.dedent(interaction_lines)
+        lines = lines.splitlines()
+
+        ff = vermouth.forcefield.ForceField(name='test_ff')
+        vermouth.ffinput.read_ff(lines, ff)
+        block = ff.blocks['GLY']
+
+        assert set(block.edges) == set(edges)
+
+    # test_interaction_fail_reference
+    @staticmethod
+    @pytest.mark.parametrize('interaction_lines', (
+        # Refers, by index, to an atom out of range
+        """
+        [ bonds ]
+        1 6
+        """,
+        # Refers, by name, to a non-defined atom (SC2 does not exist)
+        """
+        [ bonds ]
+        BB SC2
+        """,
+        # One missing atom in the definition (refers to only one atom instead of 2)
+        """
+        [ bonds ]
+        BB
+        """,
+        # Prefixed atom in a block interaction
+        """
+        [ bonds ]
+        BB +SC1
+        """,
+        """
+        [ bonds ]
+        BB -SC1
+        """,
+        """
+        [ bonds ]
+        BB <SC1
+        """,
+        """
+        [ bonds ]
+        BB >SC1
+        """,
+        # Not enough atoms for the interaction
+        """
+        [ bonds ]
+        BB
+        """,
+    ))
+    def test_interaction_fail_reference(interaction_lines):
+        """
+        Faulty atom references in interactions lead to an exception.
+        """
+        lines = """
+        [ moleculetype ]
+        GLY 1
+
+        [ atoms ]
+        1 P4 1 ALA BB 1
+        ; The following atoms have names that should not be valid
+        2 P4 2 TRP +SC1 1
+        3 P4 2 TRP -SC1 1
+        4 P4 2 TRP <SC1 1
+        5 P4 2 TRP >SC1 1
+        """
+        lines = textwrap.dedent(lines) + textwrap.dedent(interaction_lines)
+        lines = lines.splitlines()
```

Comment:
Splitlines first, add after. Otherwise you need to be very very sure the first ends in a newline character.